force-app/main/default/classes/RD2_RecurringDonations_TDTM.cls

```diff
@@ -364,41 +363,14 @@ public class RD2_RecurringDonations_TDTM extends TDTM_Runnable {
         }
 
         if (triggerAction == TDTM_Runnable.Action.AfterInsert && shouldEvaluateOppsSynchronouslyOnInsert(rds.size() > 1)) {
-            new RD2_QueueableService.EvaluateInstallmentOpportunities(rdIds, rdIdsWhereScheduleChanged)
+            new RD2_QueueableService.EvaluateInstallmentOpportunities(rdIds, rdIdsWhereScheduleChanged, oldRds)
```

Comment:
This was the best way to have the Change History created in a single place. In the Trigger context, we provide the "old list" for comparison, and in the non-Trigger context, we create our own "old list" by cloning the RDs before updates occur (see my other note)